drivers/accel/adxl38x/adxl38x.c

```diff
@@ -0,0 +1,992 @@
+ *
+ * @param dev       - The device structure.
+ *
+ * @return ret      - Result of setting to stanby mode.
+*******************************************************************************/
+static int adxl38x_set_to_standby(struct adxl38x_dev *dev)
+{
+	int ret;
+	enum adxl38x_op_mode op_mode = ADXL38x_MODE_STDBY;
+	uint8_t op_mode_reg_val;
+
+	ret = adxl38x_read_device_data(dev, ADXL38x_OP_MODE, 1, &op_mode_reg_val);
+	op_mode_reg_val = (op_mode_reg_val & ~ADXL38x_MASK_OP_MODE) | op_mode;
+	ret |= adxl38x_write_device_data(dev, ADXL38x_OP_MODE,
+					 1, &op_mode_reg_val);
+	// 10ms wait in the standby mode
+	no_os_mdelay(10);
+
+	if (!ret)
+		dev->op_mode = op_mode;
+
+	return ret;
+}
+
+
+/***************************************************************************//**
+ * @brief Executed Selftest on the sensing axes and returns the outcome of
+ * the test.
+ *
+ * @param dev   - The device structure.
+ * @param op_mode  - Operation mode in which self-test is executed
+ * @param st_x 	- Result of X-axis self test success/failure
+ * @param st_y 	- Result of X-axis self test success/failure
+ * @param st_z 	- Result of X-axis self test success/failure
+ *
+ * @return ret  - Outcome of the selftest.
+*******************************************************************************/
+int adxl38x_selftest(struct adxl38x_dev *dev, enum adxl38x_op_mode op_mode,
+		     bool *st_x, bool *st_y, bool *st_z)
+{
+	int ret;
+	uint8_t write_data;
+	uint8_t array_raw_data[6] = {0};
+	uint8_t array_rearranged_data[6] = {0};
+	int32_t low_limit_xy, low_limit_z;
+	int32_t up_limit_xy, up_limit_z;
+	int32_t st_delta_data[3] = {0};
+	int32_t st_positive_data[3] = {0};
+	int32_t st_negative_data[3] = {0};
+	int64_t x_sum = 0;
+	int64_t y_sum = 0;
+	int64_t z_sum = 0;
+	*st_x = false;
+	*st_y = false;
+	*st_z = false;
+
+	/* Enter standby mode */
+	ret = adxl38x_set_op_mode(dev, ADXL38x_MODE_STDBY);
+
+	/* Enable desired channels (except temperature) */
+	if(op_mode == ADXL38x_MODE_HRT_SND) {
+		write_data = (ADXL38x_CH_EN_X << 4);  //Only one axis
+	} else
+		write_data = (ADXL38x_CH_EN_XYZ << 4);
+	ret = adxl38x_write_device_data(dev, ADXL38x_DIG_EN, 1, &write_data);
+	if (ret)
+		return ret;
+
+	/* Enable desired OP mode */
+	ret = adxl38x_set_op_mode(dev, op_mode);
+
+	/* Enable ST mode, force and positive direction */
+	ret = adxl38x_read_device_data(dev, ADXL38x_SNSR_AXIS_EN, 1, &write_data);
+	write_data = (write_data & 0x3F) | 0xC0;
+	ret = adxl38x_write_device_data(dev, ADXL38x_SNSR_AXIS_EN, 1, &write_data);
+
+	/* Measure output on all three axes for at least 25 samples and compute average*/
+	for(int k = 0; k < 25; k++) {
+		no_os_mdelay(10);
+		ret = adxl38x_read_device_data(dev, ADXL38x_XDATA_H, 6, array_raw_data);
+		int j = 0;
+		for(int i = 0; i <  6; i+=2) {
+			array_rearranged_data[i] = array_raw_data[j];
+			array_rearranged_data[i+1] = array_raw_data[j+1];
+			j += 2;
+		}
```

Comment:
no difference between array_rearranged_data and array_raw_data. array_rearranged_data is not needed